README.md

```diff
@@ -13,3 +13,139 @@ This is an NWB extension for storing metadata of devices used in optical experim
+
+```mermaid
+%%{init: {'theme': 'base', 'themeVariables': {'primaryColor': '#ffffff', "primaryBorderColor': '#144E73', 'lineColor': '#D96F32'}}}%%
+classDiagram
+    direction BT
+    class DeviceModel{
+        <<Device>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        model : text, optional
+    }
+    class Indicator{
+        <<Device>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        label : text
+        injection_location : text, optional
+        injection_coordinates_in_mm : numeric, length 3, optional
+    }
+    class Effector{
+        <<Device>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        label : text
+        injection_location : text, optional
+        injection_coordinates_in_mm : numeric, length 3, optional
+    }
+    class OpticalFiber{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        numerical_aperture : numeric, optional
+        core_diameter_in_um : numeric, optional
+    }
+    class ExcitationSource{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        illumination_type : text, optional
+        excitation_wavelength_in_nm : numeric, optional
+        power_in_W : numeric, optional
+        intensity_in_W_per_m2 : numeric, optional
+        exposure_time_in_s : numeric, optional
+    }
+    class PulsedExcitationSource{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        peak_power_in_W : numeric, optional
+        peak_pulse_energy_in_J : numeric, optional
+        pulse_rate_in_Hz : numeric, optional
+    }
+    class Photodetector{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        detector_type : text, optional
+        detected_wavelength_in_nm : numeric, optional
+        gain : numeric, optional
+    }
+    class DichroicMirror{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        cut_on_wavelength_in_nm : numeric, optional
+        cut_off_wavelength_in_nm : numeric, optional
+        reflection_bandwidth_in_nm : numeric, optional
+        transmission_bandwidth_in_nm : numeric, optional
+        angle_of_incidence_in_degrees : numeric, optional
+    }
+    class OpticalFilter{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        filter_type : text, optional
+    }
+    class BandOpticalFilter{
```

Comment:
Doesn't this, and the `EdgeOpticalFilter`, need that field that distinguishes the type (stop or pass)? Might just be missing from the mermaid

Reply:
Should I specify in the diagram the inherited fields?